backend/core/src/listings/listings.service.ts

```diff
@@ -1,153 +1,43 @@
-        params.filter,
-        filterTypeToFieldMap,
-        innerFilteredQuery
-      )
-    }
-
-    // TODO(avaleske): Typescript doesn't realize that the `paginate` bool is a
-    // type guard, but it will in version 4.4. Once this codebase is upgraded to
-    // v4.4, remove the extra type assertions on `params.limit` below.
-    const paginate = params.limit !== "all" && params.limit > 0 && params.page > 0
-    if (paginate) {
-      // Calculate the number of listings to skip (because they belong to lower page numbers).
-      const offset = (params.page - 1) * (params.limit as number)
-      // Add the limit and offset to the inner query, so we only do the full
-      // join on the listings we want to show.
-      innerFilteredQuery.offset(offset).limit(params.limit as number)
-    }
-    const view = getView(this.listingRepository.createQueryBuilder("listings"), params.view)
-
-    let listings = await view
-      .getViewQb()
-      .andWhere("listings.id IN (" + innerFilteredQuery.getQuery() + ")")
-      // Set the inner WHERE params on the outer query, as noted in the TypeORM docs.
-      // (WHERE params are the values passed to andWhere() that TypeORM escapes
-      // and substitues for the `:paramName` placeholders in the WHERE clause.)
-      .setParameters(innerFilteredQuery.getParameters())
-      .orderBy(getOrderByCondition(params))
-      // Order by units.maxOccupancy is applied last so that it affects the order
-      // of units _within_ a listing, rather than the overall listing order)
-      .addOrderBy("units.max_occupancy", "ASC", "NULLS LAST")
-      .getMany()
-
-    // get summarized units from view
-    listings = view.mapUnitSummary(listings)
-    // Set pagination info
-    const itemsPerPage = paginate ? (params.limit as number) : listings.length
-    const totalItems = paginate ? await innerFilteredQuery.getCount() : listings.length
-    const paginationInfo = {
-      currentPage: paginate ? params.page : 1,
-      itemCount: listings.length,
-      itemsPerPage: itemsPerPage,
-      totalItems: totalItems,
-      totalPages: Math.ceil(totalItems / itemsPerPage), // will be 1 if no pagination
-    }
-
-    // There is a bug in nestjs-typeorm-paginate's handling of complex, nested
-    // queries (https://github.com/nestjsx/nestjs-typeorm-paginate/issues/6) so
-    // we build the pagination metadata manually. Additional details are in
-    // https://github.com/CityOfDetroit/bloom/issues/56#issuecomment-865202733
-    const paginatedListings: Pagination<Listing> = {
-      items: listings,
-      meta: paginationInfo,
-      // nestjs-typeorm-paginate leaves these empty if no route is defined
-      // This matches what other paginated endpoints, such as the applications
-      // service, currently return.
-      links: {
-        first: "",
-        previous: "",
-        next: "",
-        last: "",
-      },
-    }
-    return paginatedListings
+    return await this.listingRepository.list(params)
```

Comment:
To this and 
> If the idea is to pull out relationships and use separate fetches, then I think we'll have to do that work before this could get to production

I agree. I considered this to be the first step to unify retrieving listings and the next one to actually split some models. Removing joining AmiCharts from this query I think should speed things up a lot, because they are duplicated in every unit.

Reply:
Ok, how do you want to handle that? By keeping this branch going? It seems pretty straight forward to remove the ami chart relationship from units, especially since we already have the amiChartId and addUnitsSummarized already fetches the ami charts to pass into summarizeUnits.